compaction/compaction_manager.cc

```diff
@@ -679,6 +679,13 @@ compaction_manager::compaction_manager(config cfg, abort_source& as)
     }
 }
 
+compaction_manager::compaction_manager(config cfg, abort_source& as, tasks::task_manager& tm)
```

Comment:
nit: the patch subject needs to mention task manager.
otherwise, "compaction module" is too generic.
With @tchaikov's work to apply c++ modules, we may end up with a c++ compaction module which is completely unrelated to the module in this patch